File: OpenMP_cpu/adv_intro/2_timers/solutions/ans_time_print.cpp
#include  <stdio.h>
#include <unistd.h>
#include    <omp.h>
// TODO 1a: Include timer.hpp
#include "timer.hpp"
int main(){

// TODO 1a: Instantiate a Timer class as timer.
   Timer timer;

// TODO 1b: Time a 1 second sleep
   timer.start(" 1 sec sleep ");
   sleep(1);
   timer.stop();

// TODO 1c: Now time a print statement
   timer.start(" printf io   ");
   printf( " Hello Helsinki.\n");
   timer.stop();

// TODO 1d: report measurements
   timer.print();

}

File: OpenMP_cpu/adv_intro/2_timers/solutions/ans_time_print.f90

include  "timer.f90"

program main

!! TODO 1a: Use the module in timer.f90 
!!          and instantiate Timer_Collector (type) class as timer
   use mod_Timer
   type(Timer_Collector) timer

!! TODO 1b: Time a 1 second sleep
   call timer%start(" 1 sec sleep ");
      call sleep(1);
   call timer%stop();

!! TODO 1c: Now time a print statement
   call timer%start(" printf io   ");
      print*, " Hello Helsinki."
   call timer%stop();

!! TODO 1d: report measurements
   call timer%print();

end program main
